examples/flask/test_rock/app.py

```diff
@@ -16,11 +18,86 @@
 import pymysql
 import pymysql.cursors
 import redis
+from celery import Celery, Task
 from flask import Flask, g, jsonify, request
 
+
+def hostname():
+    """Get the hostname of the current machine."""
+    return socket.gethostbyname(socket.gethostname())
+
+
+def celery_init_app(app: Flask, broker_url: str) -> Celery:
+    """Initialise celery using the redis connection string.
+
+    See https://flask.palletsprojects.com/en/3.0.x/patterns/celery/#integrate-celery-with-flask.
+    """
+
+    class FlaskTask(Task):
+        def __call__(self, *args: object, **kwargs: object) -> object:
+            with app.app_context():
+                return self.run(*args, **kwargs)
+
+    celery_app = Celery(app.name, task_cls=FlaskTask)
+    celery_app.set_default()
+    app.extensions["celery"] = celery_app
+    app.config.from_mapping(
+        CELERY=dict(
+            broker_url=broker_url,
+            result_backend=broker_url,
+            task_ignore_result=True,
+        ),
+    )
+    celery_app.config_from_object(app.config["CELERY"])
+    return celery_app
+
+
 app = Flask(__name__)
 app.config.from_prefixed_env()
 
+broker_url = os.environ.get("REDIS_DB_CONNECT_STRING")
```

Comment:
Minor: no need to validate the REDIS_DB_CONNECT_STRING?

Reply:
yeah, that's a good point, not validating the url will probably make celery work even if there is no broker (which is good and bad). I would leave like it is for now (and update it in another PR if it creates problems having celery crashing when there is no redis). 
Thanks for the tip!